cohere/responses/custom_model.py

```diff
@@ -73,6 +74,23 @@ def from_dict(cls, data: Dict[str, Any]) -> "CustomModel":
             model_id=data["model"]["route"] if "model" in data else None,
         )
 
+@dataclass
+class ModelMetric(CohereObject):
+    created_at: datetime
+    step_num: int
+    loss: float
+    accuracy: float
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "ModelMetric":
+        # model metrics timestamp sometimes contains nanoseconds, so we truncate
+        dt_concat = data["created_at"][:26] + data["created_at"][-1:]
```

Comment:
this is cringy, but for some reason the timestamps back from the metrics.json is variable up to nanoseconds.. I can instead use a different library (I see dateutil.parser can handle variable seconds like this), or if anyone has a different suggestion here.

Reply:
honestly I have no better suggestion here other than at least pulling this logic into a function with a test, so it is more visible why it was done like this.

Reply:
k, I'll add the test. wanted to see if there was disagreement before continuing